aspnet/security/authorization/secure-data.rst

```diff
@@ -0,0 +1,369 @@
+
+Resource based authorization
+-----------------------------
+
+- Create an *Authorization* folder for the handlers and classes we will create to implement authorization.
+- Create a *ContactIsOwnerAuthorizationHandler* class we can invoke to verify the user acting on the resource owns the resource. Create this in the *Authorization* folder.
+
+.. literalinclude:: secure-data/samples/final/Authorization/ContactIsOwnerAuthorizationHandler.cs
+  :language: c#
+
+The ``ContactIsOwnerAuthorizationHandler`` returns ``Succeed`` if the user is the contact owner. We're not checking the requirement parameter; an owner can perform any requirement on data they own.
+
+Services using Entity Framework Core must be registered for :ref:`dependency injection <fundamentals-dependency-injection>` using :dn:method:`~Microsoft.Extensions.DependencyInjection.ServiceCollectionServiceExtensions.AddScoped`. The ``ContactIsOwnerAuthorizationHandler`` uses ASP.NET Core Identity, which is built on Entity Framework Core. Register the ``ContactIsOwner`` handler with the service collection so it will be available to the ``ContactsController`` through :ref:`dependency injection <fundamentals-dependency-injection>`. Add the following code to the end of ``ConfigureServices``:
+
+.. literalinclude::  secure-data/samples/final/Startup.cs
+  :language: c#
+  :start-after: #region snippet_AddScoped
+  :end-before: #endregion
+  :dedent: 12
+
+Update the ``ContactsController`` constructor to resolve the ``IAuthorizationService`` service so we'll have access to our authorization handlers we have registered. While we're at it we'll also get the ``Identity`` ``UserManager`` service:
+
+.. literalinclude:: secure-data/samples/final/Controllers/ContactsController.cs
+  :language: c#
+  :start-after: #region snippet_ContactsController
+  :end-before: #endregion
+  :dedent: 4
+  :emphasize-lines: 4,5,9,10,13,14
+
+Add a ``ContactOperationsRequirements`` class to the *Authorization* folder to contain the requirements our app supports:
+
+.. literalinclude:: secure-data/samples/final/Authorization/ContactOperationsRequirements.cs
+  :language: c#
+
+Update the ``HTTP POST Create`` method to add the user ID to the ``Contact`` model:
+
+.. literalinclude:: secure-data/samples/final/Controllers/ContactsController.cs
+  :language: c#
+  :start-after: #region snippet_Create
+  :end-before: #endregion
+  :dedent: 8
+  :emphasize-lines: 8
+
+Update both ``Edit`` methods to use the authorization filter to verify the user owns the contact. Add ``OwnerID`` to the ``Bind`` list:
+
+.. literalinclude:: secure-data/samples/final/Controllers/ContactsController.cs
+  :language: c#
+  :start-after: #region snippet_Edit
+  :end-before: #endregion
+  :dedent: 8
+  :emphasize-lines: 14-19,35-40,28
+
+Add the ``OwnerID`` as a hidden field so it will be available to the ``HTTP POST Edit`` method:
```

Comment:
Why? This begs for abuse. You don't add the owner ID in the source, you retrieve the contact in the controller POST method, and get the owner ID from the contact your retrieved, otherwise someone could just F12 and change the owner in the HTML.